### ISSUE TITLE
is "new_global_step = global_step + 1" necessary?

### ISSUE THREAD

**Issue by crescentluna** [google-research/bert#168]
I write a loggerHook (based on tf.train.SessionRunHook) to record the training steps and I find that the step is twice smaller than the tensor 'global_step ' printed. 

Since the tensor `global_step` will add 1 by itself in each iteration, why we need to do `new_global_step = global_step + 1` manually and assign the new_global_step value to global_step in the create_optimizer function?  It seems the global_step is **twice bigger**  than its actual value.

**Comment by yukyang** [comment]
@crescentluna Have you found any explanation about the twice bigger global_step？Right now I'm implementing gradient accumulation and confused about these lines of code as well
